File: 1950-sign-of-the-product-of-an-array/1950-sign-of-the-product-of-an-array.py
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        # reduce(func, iterable, init)
        # lambda input arguement: output expression
        return 0 if 0 in nums else reduce(lambda x, y: x * (1 if y > 0 else -1), nums, 1)

File: 1950-sign-of-the-product-of-an-array/1950-sign-of-the-product-of-an-array.cpp
class Solution {
public:
    int arraySign(vector<int>& nums) {
        int ans = 1;
        for(auto it: nums) {
            if(it < 0) {
                ans *= -1;
            }
            else if(it == 0) {
                return 0;
            }
        }
        return ans;
    }
};